lib/era_ja/conversion.rb

```diff
@@ -21,31 +26,38 @@ def to_era(format = "%o%E.%m.%d")
       @era_format.sub!(/%E/) { |m| m + ' ' }
       str_time = strftime(@era_format)
       if @era_format =~ /%E/
-        if self.to_time < ::Time.mktime(1868,9,8)
-          raise "#to_era is expeted later in 1868,9,8"
-        elsif self.to_time < ::Time.mktime(1912,7,30)
-          str_time = era_year(year - 1867, 'M')
-        elsif self.to_time < ::Time.mktime(1926,12,25)
-          str_time = era_year(year - 1911, 'T')
-        elsif self.to_time < ::Time.mktime(1989,1,8)
-          str_time = era_year(year - 1925, 'S')
-        else
-          str_time = era_year(year - 1988, 'H')
-        end
+        str_time = era_year_str(*era_year)
       end
       str_time.gsub(/%J(\d+)/) { to_kanzi($1) }
     end
+    
+    # Get Japanese year.
+    # @return [String, Fixnum] (era_alphabet, era_num)
+    def era_year
+      checked_era = nil
+      checked_era_start = nil
+      ERA.each_pair do |key, arr|
+        _ , start = arr
+        if self.to_time < start
+          raise "era_ja gem does'n t work before September 7, 1868." unless checked_era
+          break
+        end
```

Comment:
How do you think about moving logic of checking invalid time?

I think this code is a little complex.
Because there are two condition statements (` self.to_time < start` and `checked_era` is exist.), and `self.to_time < start` has two purposes.
One is whether `self` is included in era(Is 1984,1,1 included 昭和?).
Another one is checking invalid time(before 明治?).

It is needed only `self` is included in era on this loop(1984,1,1 is include 昭和).
So logic of checking invalid time will be moved to outside this loop.

``` ruby
     def era_year
+      if self.to_time < ERA['M'][1]
+        raise "era_ja gem does'n t work before September 7, 1868."
+      end
+
       checked_era = nil
       checked_era_start = nil
        ERA.each_pair do |key, arr|
         _ , start = arr
-        if self.to_time < start
-          raise "era_ja gem does'n t work before September 7, 1868." unless checked_era
-          break
-        end
+        break if self.to_time < start
         checked_era = key
         checked_era_start = start
       end
```